utils/fixed_decimal/src/decimal.rs

```diff
@@ -2078,6 +2078,185 @@ impl FixedDecimal {
     }
 }
 
+/// A struct containing a [`FixedDecimal`] significand together with an exponent, representing a
+/// number written in scientific notation, such as 1.729×10³.
+/// This structure represents any 0s shown in the significand and exponent,
+/// and an optional sign for both the significand and the exponent.
+#[derive(Debug, Clone, PartialEq)]
+pub struct ScientificDecimal {
+    significand: FixedDecimal,
+    // Invariants:
+    // - exponent.lower_magnitude == 0 (exponent has no fractional part).
+    exponent: FixedDecimal,
```

Comment:
Oh, I did not know we had that.

Reply:
To be clear, this is a "Thought", which is just a thought, not strong enough to be a Suggestion. Take it or leave it.

Reply:
I like that idea, but the name seems off: this is fixed as in fixed-point; how about IntegerDecimal? (compare ScientificDecimal rather than FixedScientific).

Reply:
I see it like BigDecimal vs BigInteger. Perhaps at some point we may want ScientificInteger. The "fixed" prefix isn't particularly meaningful except to say that we're a special kind of decimal, or a special kind of integer